src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py

```diff
@@ -200,7 +220,7 @@ def show_cache_contents(cmd, resource_group_name=None, item_name=None, resource_
     return cache_obj['_payload']
 
 
-def delete_cache_contents(cmd, resource_group_name=None, item_name=None, resource_type=None):
+def delete_cache_contents(cmd, resource_group_name, item_name, resource_type):
```

Comment:
My understanding is that the cache is most valuable for holding onto objects that have not yet been created in Azure. Once a resource has already been provisioned, it's cheap to get a reference to it. For that reason, does it make sense to do a cache delete after every PUT? At least of the objects that were the subject of that PUT?

Reply:
yes, i also had the similar though, since @tjprescott will have a separate PR to address stale data, I suggest we make more discussions over that one.

Reply:
I think it ends up being a wash either way. Right now I always cache even when doing a PUT so that a subsequent call, when it reaches into the cache, will contain the same change as the PUT. It would be perfectly viable to ensure the PUT was successful and then delete the cached object. The next call would then be a miss and it would pull from Azure and continue.

Reply:
Sounds good to me. In my current image builder examples that use the cache, I add a final command to delete the cache object just so users of the command are aware.